Control coverage/Feature/ApplicationGateway.md

```diff
@@ -0,0 +1,108 @@
+
+	 Use the Azure portal to configure WAF on the Application Gateway.
+
+- **PowerShell** 
+
+	 ```powershell 
+
+	# Below commands will be useful to Configure WAF on Application Gateway
+    Connect-AzAccount
+	Set-AzContext -SubscriptionId "<sub id>"`
+	#Get Application Gateway and existing policy object
+	$appgw = Get-AzApplicationGateway -Name "applicationgatewayName" -ResourceGroupName   "RgName"
+	$policy = Get-AzApplicationGatewayFirewallPolicy -Name "WAFPolicyName" -ResourceGroupName "RgName"
+		
+	#Attach the policy to an Application Gateway
+	$appgw.FirewallPolicy = $policy		
+	#Save the Application Gateway
+	Set-AzApplicationGateway -ApplicationGateway $appgw
+
+	#Below commands will be useful to configure the WAF at the listener level in the Application Gateway:  
+	Connect-AzAccount
+	Set-AzContext -SubscriptionId "<sub id>"`
+	#Get Application Gateway, Listener and existing policy object
+	$appgw = Get-AzApplicationGateway -Name "applicationgatewayName" -ResourceGroupName   "RgName"
+	$policy = Get-AzApplicationGatewayFirewallPolicy -Name "WAFPolicyName" -ResourceGroupName "RgName"
+	$listener = Get-AzApplicationGatewayHttpListener -Name "L1" -ApplicationGateway $appgw
+	#Attach the policy to an Application Gateway Listener
+	$listener.FirewallPolicy = $policy
+		
+	#Save the Application Gateway Listener
+	Set-AzApplicationGatewayHttpListener  -FirewallPolicy  $policy -ApplicationGateway $appgw
+
+	Below commands could be run to change the Policy Mode to Prevention mode:
+	Connect-AzAccount
+	Set-AzContext -SubscriptionId "<sub id>"`
+	#Get Application Gateway Firewall policy
+	$policy = Get-azapplicationGatewayFirewallPolicy -Name "WAFPolicyName" -ResourceGroupName "RGName"
+	#Get the Policy Settings and Set the Mode to prevention
+	$Policysettings = $policy.PolicySettings
+	$Policysettings.Mode = "Prevention"
+	#Save the WAF Policy
+	Set-AzApplicationGatewayFirewallPolicy -PolicySetting $Policysettings -InputObject $policy
+
+	Run command 
+	Retrieve-ApplicationGatewayVirtualNetworkDDoSNotConfigured - to retrieve the list of Application Gateway virtual network where DDoS is not enabled.
+	Run Enable-DDoSProtectionPlanOnVirtualNetwork to remediate the Virtual Network(s) retrieved from above command. 
+
+	Run command Retrieve-ApplicationGatewaySubnetNSGNotConfigured to retrieve the list of Application Gateway subnet where NSG is not configured.
+	Run Add-NSGConfigurationOnSubnet to remediate the Subnet(s) retrieved from above command. 
+
+     # For more help run:
+	 Get-Help Retrieve-ApplicationGatewayVirtualNetworkDDoSNotConfigured -Detailed
+	 Get-Help Enable-DDoSProtectionPlanOnVirtualNetwork -Detailed
+	 Get-Help Retrieve-ApplicationGatewaySubnetNSGNotConfigured -Detailed
+	 Get-Help Add-NSGConfigurationOnSubnet -Detailed
+	 ```  
+
+### Azure Policy or ARM API used for evaluation 
+
+- ARM API to list all Application Gateway: /subscriptions/{0}/providers/Microsoft.Network/applicationGateways?api-version=2022-01-01<br />
```

Comment:
Are we getting everything we need for control evaluation by just single API call?